Rb.Forms.Barcode.iOS/BarcodeScannerRenderer.cs

```diff
@@ -0,0 +1,206 @@
+﻿using Xamarin.Forms;
+using Rb.Forms.Barcode.Pcl;
+using Rb.Forms.Barcode.iOS;
+using Xamarin.Forms.Platform.iOS;
+using AVFoundation;
+using UIKit;
+using CoreGraphics;
+using Foundation;
+using System.ComponentModel;
+using CoreFoundation;
+using RebuyBarcode = Rb.Forms.Barcode.Pcl.Barcode;
+using Rb.Forms.Barcode.iOS.Extensions;
+using System.Diagnostics;
+using Rb.Forms.Barcode.Pcl.Logger;
+using Rb.Forms.Barcode.iOS.Logger;
+
+[assembly: ExportRenderer(typeof(BarcodeScanner), typeof(BarcodeScannerRenderer))]
+namespace Rb.Forms.Barcode.iOS
+{
+    public class BarcodeScannerRenderer : ViewRenderer, IAVCaptureMetadataOutputObjectsDelegate, ILog
+    {
+        private NSObject orientationObserverToken;
+
+        private static Configuration configuraiton;
+        private AVCaptureSession session;
+        private AVCaptureDevice device;
+        private AVCaptureDeviceInput input;
+        private AVCaptureMetadataOutput output;
+        private AVCaptureVideoPreviewLayer captureVideoPreviewLayer;
+        private UIView view;
+        private BarcodeScanner barcodeScanner;
+
+        public static void Init(Configuration configuraiton)
+        {
+            BarcodeScannerRenderer.configuraiton = configuraiton;
+        }
+
+        [Export("captureOutput:didOutputMetadataObjects:fromConnection:")]
+        public void DidOutputMetadataObjects(AVCaptureMetadataOutput captureOutput, AVMetadataObject[] metadataObjects, AVCaptureConnection connection)
+        {
+            if (!barcodeScanner.BarcodeDecoder) {
+                return;
+            }
+
+            foreach (var metadata in metadataObjects) {
```

Comment:
The barcode reader only give you the barcode types you defined in the config. no others will be scanned. yes i will break after the first one.